### DISCUSSION TITLE
havy client site memory use in non ssr application / slow application

### DISCUSSION THREAD

**Discussion by FinnThorwarth** [nuxt/nuxt#25675] [Questions]
Hey,
i have a non ssr application and i am running into performance troubles.

My app is loading a lot of data (like 900 objects with person data) from a database and i am storing the data inside the local storage with pinia.

I have different views where i show the filtered based on rules. (filtering is client site).

This worked well up to like 400-600 objects but now the application slows down over time of use, means the first operations are smooth and then is slows down. Chrome is showing memory use of up to 1GB and more.

I use like 23 different Pinia Stores.  I have no experience in how this many stores effecting the performance.

Where to start optimizing the performance?

best regards

mybe helpfull the dependencies:

 "@pinia/nuxt": "^0.4.6",
    "@tailwindcss/forms": "^0.5.3",
    "@tiptap/pm": "^2.0.0-beta.217",
    "@tiptap/starter-kit": "^2.0.0-beta.217",
    "@tiptap/vue-3": "^2.0.0-beta.217",
    "@vueup/vue-quill": "^1.1.0",
    "@vueuse/nuxt": "^9.11.0",
    "apexcharts": "^3.41.0",
    "autoprefixer": "^10.4.13",
    "cookie-universal-nuxt": "^2.2.2",
    "floating-vue": "^2.0.0-beta.20",
    "nuxt-icon": "^0.2.10",
    "pinia": "^2.0.29",
    "postcss": "^8.4.21",
    "tailwindcss": "^3.2.4",
    "tippy.js": "^6.3.7",
    "universal-cookie": "^4.0.4",
    "vue-cookies": "^1.8.2",
    "vue-tel-input": "^6.0.5",
    "vue-tippy": "^6.0.0-alpha.65",
    "vue3-apexcharts": "^1.4.1"

**Comment by typed-sigterm** [comment]
That is *memory leak*, which is a **very** troublesome problem. May be caused by closures or event listeners that forget to uninstall, etc.
I use `location.reload()` to solve it (I'm not an expert XD).

Additionally, if you integrate frameworks like [tauri](https://tauri.app/), `window.gc()` is helpful.

**Reply by FinnThorwarth** [reply]
Thanks for the hint about the location.reload that could help to improve the performance on the short run, but is not solving the problem. I managed to reduce the size of date by redesigning the api what makes it a lot faster for now. But need to find this memory leaks.
